src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PileupDetectionArgumentCollection.java

```diff
@@ -0,0 +1,92 @@
+     *
+     * NOTE: --pileup-detection is a beta feature. Use this mode at your own risk.
+     */
+    @Argument(fullName= PILEUP_DETECTION_LONG_NAME, doc = "If enabled, the variant caller will create pileup-based haplotypes in addition to the assembly-based haplotype generation.", optional = true)
+    public boolean usePileupDetection = false;
+
+    /**
+     * Enables detection of indels from the pileups in. (EXPERIMENTAL FEATURE)
+     */
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_ENABLE_INDELS, doc = "If enabled, pileup detection code will attempt to detect indels missing from assembly. (Requires `--pileup-detection` argument)", optional = true)
+    public boolean detectIndels = false;
+
+
+    /**
+     * Percentage of reads required to supprot the alt for a variant to be considered
+     */
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_SNP_THRESHOLD, doc = "Percentage of alt supporting reads in order to consider alt SNP", optional = true)
+    public double snpThreshold = 0.1;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_INDEL_THRESHOLD, doc = "Percentage of alt supporting reads in order to consider alt indel", optional = true)
+    public double indelThreshold = 0.5;
+
+
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_ABSOLUTE_ALT_DEPTH, doc = "Absolute number of alt reads necessary to be included in pileup events", optional = true)
+    public double pileupAbsoluteDepth = 0;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_INDEL_SNP_BLOCKING_RANGE, doc = "Filters out pileup snps within this many bases of an assembled indel", optional = true)
+    public int snpAdajacentToAssemblyIndel = 5;
+
+
+
+
+    /**
+     * Arguments related to the "bad read filtering" where alleles that are supported primarily by reads that fail at least one of a number of heuristics will be filtered out
+     */
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_BAD_READ_RATIO_LONG_NAME, doc = "Thresold of Alt reads rejected by bad reads heuristics to allow the variant", optional = true)
+    public double badReadThreshold = 0.0;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_PROPER_PAIR_READ_BADNESS_LONG_NAME, doc = "Reject Alt reads not in proper-pairs", optional = true)
+    public boolean badReadProperPair = true;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_EDIT_DISTANCE_BADNESS_LONG_NAME, doc = "Reject alt reads with greater than this percent edit distance from the reference", optional = true)
+    public double badReadEditDistance = 0.08;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_CHIMERIC_READ_BADNESS_LONG_NAME, doc = "Reject reads that are chimeric or supplementary", optional = true)
+    public boolean badReadSecondary = true;
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_TLEN_MEAN_LONG_NAME, doc = "Mean template length to consider for read badness. Requires '--"+PILEUP_DETECTION_TLEN_STD_LONG_NAME+"' to also be set.", optional = true)
```

Comment:
I'd use TLEN in the doc, since I found myself to be confused about what is meant by "template" for a few seconds. 

you can add a parenthetical note with "Template LENgth" or something.